oauth2_http/java/com/google/auth/oauth2/OAuth2Utils.java

```diff
@@ -94,17 +94,6 @@ static boolean headersContainValue(HttpHeaders headers, String headerName, Strin
   }
 
   /**
-   * Throw the exception with the specified cause.
-   *
-   * <p>Needed instead of constructor version to be compatible with JDK 1.5 which is required
-   * until Andriod libraries can update to JDK 1.6
-   */
-  static <T extends Throwable> T exceptionWithCause(T exception, Throwable cause) {
```

Comment:
Removing this seems like good cleanup. It was needed before because of Android constraints which seem to be more relaxed now. It seems tangential to this change. Perhaps factor this into a separate PR?